include/seastar/util/shared_token_bucket.hh

```diff
@@ -200,7 +200,7 @@ public:
     // Estimated time to process the given amount of tokens
     // (peer of accumulated_in helper)
     rate_resolution duration_for(T tokens) const noexcept {
-        return rate_resolution(tokens / _replenish_rate);
+        return rate_resolution(double(tokens) / _replenish_rate);
```

Comment:
how about:

```suggestion
        return rate_resolution{tokens} / _replenish_rate;
```

so we don't need to hardwire to double after defining `rate_resolution`, which intends to abstract the representation of the rate resolution.

Reply:
I'd say that _replenish_rate needs to be a double (and then maybe we can make Period = 1).

I'll merge this for now as a minimal fix and we can improve it later.